harvester_e2e_tests/scenarios/test_create_vm.py

```diff
@@ -219,6 +220,51 @@ def test_create_update_vm_machine_type(request, admin_session, image, keypair,
                 utils.delete_vm(request, admin_session,
                                 harvester_api_endpoints, vm_json)
 
+
+def test_create_update_vm_user_data(request, admin_session, image, single_vm,
+                                    harvester_api_endpoints,
+                                    user_data_with_guest_agent, network_data):
+    # make sure the VM instance is successfully created
+    vm_instance_json = utils.lookup_vm_instance(
+        admin_session, harvester_api_endpoints, single_vm)
+    vm_name = single_vm['metadata']['name']
+    previous_uid = vm_instance_json['metadata']['uid']
+    vol_data = single_vm['spec']['template']['spec']['volumes']
+    assert 'cloudInitNoCloud' not in vol_data
+    userdata = user_data_with_guest_agent.replace('\\n', '\n')
+    netdata = network_data.replace('\\n', '\n')
+    user_data = {
+        'cloudInitNoCloud': {
+            'networkData': netdata,
+            'userData': userdata
+        },
+        'name': 'cloudinitdisk'
+    }
+    vol_data.append(user_data)
+    resp = utils.poll_for_update_resource(
+        request, admin_session,
+        harvester_api_endpoints.update_vm % (vm_name),
+        single_vm,
+        harvester_api_endpoints.get_vm % (vm_name),
+        use_yaml=True)
+    # restart the VM instance for the updated machine type to take effect
+    utils.restart_vm(admin_session, harvester_api_endpoints, previous_uid,
+                     vm_name, request.config.getoption('--wait-timeout'))
+    resp = admin_session.get(harvester_api_endpoints.get_vm % (
+        vm_name))
+    updated_vm_data = resp.json()
+    updated_vol_data = updated_vm_data['spec']['template']['spec']['volumes']
+    found = False
+    for i in updated_vol_data:
+        if i['name'] == 'cloudinitdisk':
+            if (i['cloudInitNoCloud']['networkData'] == netdata and
+                    i['cloudInitNoCloud']['userData'] == userdata):
+                found = True
+                break
+
+    assert found, 'Failed to update VM %s to add userDate & NetworkData' % (
+        vm_name)
```

Comment:
We should reboot the VM to ensure the changes works by successfully SSH into the VM.